src/platform/cc13x2_26x2/BLEManagerImpl.cpp

```diff
@@ -1803,6 +1812,55 @@ void BLEManagerImpl::CHIPoBLEProfile_charValueChangeCB(uint8_t paramId, uint16_t
     }
 }
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+/*********************************************************************
+ * @fn      CHIPoBLEProfile_c3ValueReadCB
+ *
+ * @brief   Callback from CHIPoBLE Profile indicating a commissioner
+ *          reading an additionald commissioning-related data (C3) from
+ *          commissionee.
+ *          Calling context (BLE Stack Task)
+ *
+ * @param   destBuf - buffer to copy data to
+ *          maxLen - maximum length of destBuf
+ *
+ * @return  None.
+ */
+void BLEManagerImpl::CHIPoBLEProfile_c3ValueReadCB(uint8_t * destBuf, uint16_t maxLen, uint16_t connHandle)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    System::PacketBufferHandle packetBuf;
+    uint16_t len;
+
+    char serialNumber[ConfigurationManager::kMaxSerialNumberLength + 1];
+    uint16_t lifetimeCounter = 0;
```

Comment:
lifetimeCounter should be part of the `#if`. This means that `generateAdditionalDataPayload` should take some `AdditionalDataPayloadGeneratorParams` (which it seems to do according to current master: https://github.com/project-chip/connectedhomeip/blob/master/src/setup_payload/AdditionalDataPayloadGenerator.h#L79):


```cpp
CHIP_ERROR generateAdditionalDataPayload(AdditionalDataPayloadGeneratorParams & params,
                                             chip::System::PacketBufferHandle & bufferHandle,
                                             BitFlags<AdditionalDataFields> additionalDataFields);
```

Reply:
It seems this needs merge with master and updates.

Reply:
Yes, you're right, this function call is stale.
I will address this on my next re-base (as the build error will show up).